### PR TITLE
Updates to register page, improved error handling

### PR DESCRIPTION
**Pull request by Jatewo**
## Description

Implemented a new error handling system, meant to be used across the platform, aswell as split name in User model. Added LogOutButton with tests for logout

## Changes
- added LogOutButton to profile page
- unit tests added for LogOutButton by comparing to snapshot
- snapshot for LogOutButton added
- unit tests for logout removing the token, and navigating
- modified styling to nativewind for LogOutButton
- Translations added in LogOutButton
- Cleanup in all touched files
- Documentation for all touched files
- Documentation on all translations (pending proper translations from PO) both in loginForm.js and registerForm.js
- Refactored console.log in registerUser in userApi 
- Moved frontend-only validation to own file
- Added errorCodes.js, that contains all current JSON error objects
- Deleted displayPasswordGuidelines function from formTextField.js
- Deleted all traces of isFontLoaded in files
- Refactored all errors in loginForm.js to new error handling system
- Refactored registerForm to use first name and last name instead of a single name
- Added validation to name and email in registerForm.js
- Moved regex check from registerForm to validation.js
- Refactored the overall validation check (the one that checks everything) to better comply with codeScene standards for complex methods (cyclomatic complexity)
- Refactored all errors in registerForm.js to new error handling system.

## Related Issues

None documented.

## Checklist

- [X] Code has been tested locally and passes all relevant tests.
- [X] Documentation has been updated to reflect the changes, if applicable.
- [X] Code follows the established coding style and guidelines of the project.
- [X] All new and existing tests related to the changes have passed.
- [X] Any necessary dependencies or new packages have been properly documented.
- [X] Pull request title and description are clear and descriptive.
- [X] Reviewers have been assigned to the pull request.
- [X] Any potential security implications have been considered and addressed.
- [X] Performance impact of the changes has been evaluated, if relevant.

## Screenshots (if applicable)

None relevant

## Notes for Reviewers

For future reference, CERT group hopes that the rest of the teams will adopt our new error handling system.

### CODE REVIEW

File: eml/api/userApi.js [hunk 1]
```diff
@@ -21,23 +20,44 @@ export const client = axios.create({
...
+    } else {
+      throw e;
+    }
+  }
 }
 
+/**
+ * Sends a request to the backend to login an existing user.
+ * @param {Object} obj should contain the following properties:
+ * - email
```
Comment (magnemo) [inline eml/api/userApi.js:50]: Hvad er  email, password

File: eml/api/userApi.js [hunk 2]
```diff
@@ -21,23 +20,44 @@ export const client = axios.create({
...
+ * - password
+*/
 export const loginUser = async (obj) => {
   const res = await client.post('/api/auth/login', obj)
   return res.data
 }
-
+/**
+ * Sends a request to the backend to delete a user.
+ * @param {user_id} user_id
```
Comment (magnemo) [inline eml/api/userApi.js:59]: Kan i ikke forklare parameter

File: eml/components/login/RegisterForm.js [hunk 3]
```diff
@@ -36,107 +41,144 @@ export default function RegisterForm() {
...
-      && passwordContainsLetter && confirmPasswordAlert === "") {
-      setIsAllInputValid(true);
-    } else {
-      setIsAllInputValid(false);
-    }
-  }
-
   /**
    * Stores the user info in async storage
    * @param {*} id user id
```
Comment (magnemo) [inline eml/components/login/RegisterForm.js:171]: Mangler typer

File: eml/components/login/RegisterForm.js [hunk 4]
```diff
@@ -36,107 +41,144 @@ export default function RegisterForm() {
...
-   * @param {*} realName 
+   * @param {*} firstName 
+   * @param {*} lastName
    * @param {*} email 
    */
-  async function createProfile(id, realName, email) {
+  async function saveUserInfoLocally(id, firstName, lastName, email) {
```
Comment (magnemo) [inline eml/components/login/RegisterForm.js:176]: Beskrivelse

File: eml/api/userApi.js (hunk 1 above)
Comment (JackSMunn) [reply eml/api/userApi.js:50]: Objektet som den tager ind som parameter, skal have 2 fields, email og password

**Pull request merged**